### PR TITLE
feat(lib): add disabled state for select item

### LINKED ISSUE #71: Is there a way to programatically enable/disable an element as selectable?
Just to be clear, I'm not asking about changing something from selected to not selected and back again. Rather, I want to make specific element selectable/not selectable via code.

According to the docs, you make a DOM element "Selectable" by wrapping all the items you want to be selectable in a "<dts-select-container>" element, and then you also need to add the [dtsSelectItem]="item" tag to the element(s) in question. 

What I'd like to do is select some elements, and then make them un-selectable again via code. How can I achieve this?

I'm curious how the tagging works... for example, if I remote the "dts-select-item" class from the element, will that be enough to make it "not selectable"? Or is there a better way of doing this that won't break ngx-drag-to-select?"

### PR DESCRIPTION
**Pull request by ennjin**
Closes #71

### CODE REVIEW

**Comment by ennjin** [comment]
@d3lm Hey. Can you take a look at this, please?

File: README.md [hunk 1]
```diff
@@ -468,16 +468,22 @@ The `dtsSelectItem` directive is used to mark DOM elements as selectable items.
 
 **Inputs**
 
-| Input         | Type | Default            | Description                                  |
-| ------------- | ---- | ------------------ | -------------------------------------------- |
-| dtsSelectItem | any  | Directive Instance | Value that is used when the item is selected |
+| Input         | Type     | Default            | Description                                  |
+| ------------- | -------- | ------------------ | -------------------------------------------- |
+| dtsSelectItem | any      | Directive Instance | Value that is used when the item is selected |
+| dtsDisabled   | boolean  | false              | Disable selection for item                   |
```
Comment (d3lm) [inline README.md:474]: I'd like to change the description to something like `Whether the item is selectable or not`.

File: projects/ngx-drag-to-select/src/lib/select-container.spec.ts [hunk 2]
```diff
@@ -108,6 +112,21 @@ describe('SelectContainerComponent', () => {
...
+
+      selectContainerInstance.select.subscribe((items) => {
+        expect(testComponent.selectedItems.length).toBe(result.length);
+        expect(items).toEqual(result);
+        expect(testComponent.selectedItems).toEqual(result);
+        done();
+      });
+
+      testComponent.getByIndex(1).dtsDisabled = true;
+      selectContainerInstance.selectAll();
```
Comment (d3lm) [inline projects/ngx-drag-to-select/src/lib/select-container.spec.ts:128]: Please add a newline before.

File: src/app/app.component.html [hunk 3]
```diff
@@ -22,6 +22,7 @@ <h3>
     <mat-checkbox data-cy="selectOnDrag" [(ngModel)]="selectOnDrag">Select on Drag</mat-checkbox>
     <mat-checkbox data-cy="dragOverItems" [(ngModel)]="dragOverItems">Drag Over Items</mat-checkbox>
     <mat-checkbox data-cy="disable" [(ngModel)]="disable">Disable</mat-checkbox>
+    <mat-checkbox data-cy="useDisableCondition" [(ngModel)]="useDisableCondition">Disable condition (even)</mat-checkbox>
```
Comment (d3lm) [inline src/app/app.component.html:25]: For the text of the checkbox let's just say `Disable even items`.

Comment (d3lm) [inline src/app/app.component.html:25]: Also could you change `useDisableCondition` to `disableEvenItems`

**Review by d3lm — CHANGES_REQUESTED** [review]
Thanks for the PR! Very much appreciated. Left a few small comments. Otherwise it looks good 👍

**Comment by ennjin** [comment]
Hi. I've updated the notes above :)

**Comment by ennjin** [comment]
There're latest changes from master branch.

File: projects/ngx-drag-to-select/src/lib/select-item.directive.ts [hunk 4]
```diff
@@ -34,6 +34,10 @@ export class SelectItemDirective implements OnInit, DoCheck {
 
   @Input() dtsSelectItem: any | undefined;
 
+  @HostBinding('class.dts-disabled')
+  @Input()
```
Comment (d3lm) [inline projects/ngx-drag-to-select/src/lib/select-item.directive.ts:38]: It's a nitpick but can you please change the order here
```suggestion
  @Input()
  @HostBinding('class.dts-disabled')
```

**Review by d3lm — CHANGES_REQUESTED** [review]
This looks good. One last thing, could you also add an e2e test for this? 🙏

**Comment by ennjin** [comment]
Updated!

File: cypress/integration/styling-and-classes.spec.ts [hunk 5]
```diff
@@ -104,4 +104,13 @@ describe('Styling and Classes', () => {
         });
     });
   });
+
+  it.only('should add `.dts-disabled` class to disabled items', () => {
```
Comment (d3lm) [inline cypress/integration/styling-and-classes.spec.ts:108]: Nitpick: I have used single quotes in the description of other tests so I'd like to streamline this:
```suggestion
  it.only(`should add 'dts-disabled' to disabled items`, () => {
```

**Review by d3lm — CHANGES_REQUESTED** [review]
Awesome work! One last small thing.

File: cypress/integration/styling-and-classes.spec.ts [hunk 6]
```diff
@@ -104,4 +104,13 @@ describe('Styling and Classes', () => {
         });
     });
   });
+
+  it(`should add '.dts-disabled' class to disabled items`, () => {
```
Comment (d3lm) [inline cypress/integration/styling-and-classes.spec.ts:108]: Just say `'dts-disabled'`, that's what I did for all the other tests as well.
```suggestion
  it(`should add 'dts-disabled' class to disabled items`, () => {
```

**Review by d3lm — CHANGES_REQUESTED** [review]

**Review by d3lm — APPROVED** [review]

**Comment by d3lm** [comment]
Merged! Thanks a lot for this PR 🙏

**Comment by ennjin** [comment]
@d3lm Thanks! I'm wondering if could you bump the new changes to the NPM?

**Comment by d3lm** [comment]
Definitely. I am trying to push on the other PR too so I can cut a single release that includes both new features.

**Comment by d3lm** [comment]
It's this one #129.

**Comment by d3lm** [comment]
I am not hearing anything, so maybe we can cut a release already. Alternatively you could pick up the PR and we can get it over the finish line.

**Comment by ennjin** [comment]
> I am not hearing anything, so maybe we can cut a release already

Yep, it would be nice if we released it

**Comment by ennjin** [comment]
So, will you bump the new version?

**Comment by d3lm** [comment]
It's now published. You can grab 4.2.0.

**Pull request merged**